base_contract/models/agreement.py

```diff
@@ -0,0 +1,55 @@
+# © 2017 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import fields, models
+
+
+class Agreement(models.Model):
+    _name = "agreement"
+    _description = "Agreement"
+
+    code = fields.Char(required=True, copy=False)
+    name = fields.Char(required=True)
+    partner_id = fields.Many2one(
+        "res.partner",
+        string="Partner",
+        ondelete="restrict",
+        domain=[("parent_id", "=", False)],
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        default=lambda self: self.env["res.company"]._company_default_get(),
```

Comment:
`_company_default_get()` should be removed: https://github.com/odoo/odoo/blob/58492733b81073694a7e2f5e790b31983f91df7d/odoo/addons/base/models/res_company.py#L205

It shows a warning at loading of the module.

```suggestion
        default=lambda self: self.env.company.id,
```